### ISSUE TITLE
BatOptParse: %prog ist replaced only once

### ISSUE THREAD

**Issue by OnkV** [ocaml-batteries-team/batteries-included#965]
* Batteries 3.1.0
* OCaml 4.11.1
* Debian 10

The documentation for `BatOptParse.OptParser` promises:

```
val make : ?⁠usage:string -> ?⁠description:string -> ?⁠version:string
    -> ?⁠suppress_usage:bool -> ?⁠suppress_help:bool
    -> ?⁠only_leading_opts:bool -> ?⁠prog:string
    -> ?⁠formatter:Formatter.t -> unit -> t
Creates a new option parser with the given options.

parameter usage
  Usage message. The default is a reasonable usage message for most programs.
  Any occurrence of the substring "%prog" in usage is replaced with the name of the
  program (see prog).
[...]
```

But only the first occurrence of `%prog` is substituted.

See file ` batteries-included/src/batOptParse.ml`:

```
let unprogify optparser s =
    (snd (BatString.replace ~str:s ~sub:"%prog" ~by:optparser.op_prog))
```

`BatString.replace` only replaces the first occurrence of the substitution string.

IMHO either the documentatioin or the implementation has to receive an update.  Thank you!

**Comment by UnixJunkie** [comment]
Can you send a fix?

**Comment by UnixJunkie** [comment]
It's pretty lame, but we don't even have a BatString.replace_all...

**Comment by UnixJunkie** [comment]
Ok, replace_all is named nreplace.

**Comment by UnixJunkie** [comment]
I sent a PR so that the code does what the documentation says.
If any maintainer is not OK with that, please raise your hand soon.

**Comment by UnixJunkie** [comment]
fixed by #966 
The ChangeLog needs to be updated.
